test/javabind.cpp

```diff
@@ -468,6 +468,9 @@ JAVA_EXTENSION_MODULE()
         .function<StaticSample::pass_value<int16_t>>("pass_short")
         .function<StaticSample::pass_value<int32_t>>("pass_int")
         .function<StaticSample::pass_value<int64_t>>("pass_long")
+        .function<StaticSample::pass_value<uint8_t>>("pass_unsigned_byte")
+        .function<StaticSample::pass_value<uint16_t>>("pass_unsigned_short")
+        .function<StaticSample::pass_value<uint32_t>>("pass_unsigned_int")
```

Comment:
I assume we would like to keep these function registrations even if we have unsigned integer support disabled. However, instead of `StaticSample::pass_value<T>`, we could register a function that simply throws a well-known exception that unit tests in Java could catch and ignore. This would make Java unit tests work regardless of the preprocessor defines the the C++ library is compiled with.

Reply:
I have removed the unit tests for the unsigned values so that the unit tests work with and without widening enabled.

The ArgType for unsigned ints isn't defined when widening is disabled so we can't define functions that accept return unsigned integers because this would trigger a compilation error.

Reply:
Cool. I would still want to have a few unit tests around unsigned integers and widening conversions but it definitely needs some updates to the test framework. I will make these changes as time permits.